Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -3315,24 +3315,34 @@ private void OnIbAutomaterExited(object sender, ExitedEventArgs e)
             if (!result.HasError && !_isDisposeCalled)
             {
                 // IBGateway was closed by IBAutomater because the auto-restart token expired or it was closed manually (less likely)
-                Log.Trace("InteractiveBrokersBrokerage.OnIbAutomaterExited(): IBGateway close detected, restarting IBAutomater in 10 seconds...");
+                Log.Trace("InteractiveBrokersBrokerage.OnIbAutomaterExited(): IBGateway close detected, restarting IBAutomater...");
 
-                Thread.Sleep(TimeSpan.FromSeconds(10));
+                Task.Factory.StartNew(() =>
+                {
+                    try
+                    {
+                        // during weekends wait until one hour before FX market open before restarting IBAutomater
+                        var delay = _ibAutomater.IsWithinWeekendServerResetTimes()
+                            ? GetNextWeekendReconnectionTimeUtc() - DateTime.UtcNow
+                            : TimeSpan.FromSeconds(10);
```

Comment:
> : TimeSpan.FromSeconds(10);

In IbAutomater we were waiting 5 minutes in some cases https://github.com/QuantConnect/IBAutomater/pull/59/files?file-filters%5B%5D=.csproj&file-filters%5B%5D=.java#diff-695a037aa05803e44f5c5815029d8faaf767e0085d76f9af26c42d8c341e1a1fL350 or a minute in others https://github.com/QuantConnect/IBAutomater/pull/59/files?file-filters%5B%5D=.csproj&file-filters%5B%5D=.java#diff-695a037aa05803e44f5c5815029d8faaf767e0085d76f9af26c42d8c341e1a1fL419 - Should we always wait only 10 seconds here?


Nit - maybe could directly use Task.Delay